src/databricks/labs/ucx/progress/workflows.py

```diff
@@ -0,0 +1,113 @@
+from databricks.labs.ucx.contexts.workflow_task import RuntimeContext
+from databricks.labs.ucx.framework.tasks import Workflow, job_task
+
+
+class MigrationProgress(Workflow):
+    """Experimental workflow that rescans the environment to reflect and track progress that has been made.
+
+    This is a subset of the assessment workflow and covers:
+
+     - Clusters
+     - Grants
+     - Jobs
+     - Pipelines
+     - Policies
+     - Tables
+     - TableMigrationStatus
+     - UDFs
+    """
+
+    def __init__(self) -> None:
+        super().__init__('migration-progress-experimental')
+
+    @job_task
+    def crawl_tables(self, ctx: RuntimeContext) -> None:
+        """Iterates over all tables in the Hive Metastore of the current workspace and persists their metadata, such
+        as _database name_, _table name_, _table type_, _table location_, etc., in the table named
+        `$inventory_database.tables`. The metadata stored is then used in the subsequent tasks and workflows to, for
+        example, find all Hive Metastore tables that cannot easily be migrated to Unity Catalog."""
+        ctx.tables_crawler.snapshot(force_refresh=True)
```

Comment:
use py4j implementation, SQL implementation is extremely slow

Reply:
@nfx: I agree it should use that crawler. That said:

 - On #2579 this property is being updated so that `tables_crawler` will be the py4j implementation.
 - The py4j implementation on `main` doesn't seem to work properly at the moment: it produces lots of errors while fetching metadata for tables, and the errors in turn trigger serialisation errors in the table-crawling task. (Integrating the changes from #2579 doesn't seem to resolve these problems.)

If this is the only remaining issue I would prefer to solve the py4j integration in a separate PR.

Reply:
ok